polaris-react/src/components/Text/Text.scss

```diff
@@ -3,6 +3,10 @@
 .root {
   margin: 0;
   text-align: inherit;
+
+  #{$se23} & {
+    max-width: 80ch;
```

Comment:
How does this look on areas of the page that are wider than 80ch? I'm thinking banners and other informational bars that stretch the screen width, or IndexTable rows that might have long bits of text in them, etc.

What happens when the `truncate` option is applied? Is cutting it off at 80ch width the desired effect, or should it stay 100% width and cut off at the edge of the container?

What about when `alignment="end"` is applied? The text container would become 80ch wide, but the text is then right aligned to the now-thinner container instead of reaching to the edge of the previously 100% width container. Similarly for `alignment="center"`; the "center" would be off. [Here's a quick demo](https://polaris.shopify.com/sandbox?code=N4Igxg9gJgpiBcIA8AhCAPABABwIZSgEsA7AcwF4AdEAFmoD5LjNMkBhXAJykeZdYBqMTgBdCYXABsAyiNxgA1plK5sVEADYGTfv1QYc%2BImXV0QmAEYRuw2QE9JMdQGcIkwlGqXrsTgHUPEQALdQBGLysbTgAlfEIAV2d1ACYI%2BQVSTgh44ih1C1IAWnkxADcYbT5dFiQAFRh0EXoASQByAFtMXFZIWHoAMkkRAG56xv7SEaQAel6Yehmxpp1qmqWu91JidphiEXVdzxAWjq6e6HnBkfWpQi2dvYPc6gmp2YuF6aXeVbWGkQ2d22u321DAIOEDDanW6SDmAyGo3%2BgPuIPU4L2kJAr2GM3hi3%2BP1%2BdWRIk4OQkInm0LOcI%2BVyRjUwZIpuCpOLxHwAdJg2BB2u0cuI2YQIMxCM5MMQIADsBBnM5CBZHDzakEYJhnEF4iIRI5MEFcJLpTgIAB3YQ8gCqirIzPVyk4uFKhDkYjFUhw8UkkkwEAAZmdnHJOMyIJr3GQtTLLOlMCRmYQdgB%2BTB%2BDVcDWkCAkUhhzCyLgWI0agCyYpE2Z2nDsmH91kwAFEyCQYMJc5hIMRnN63aLuzgsu1sFSoJZa%2B0SA3ODB8IVsgDcNL2rcYM4eQAFLIWNcG50ameQGxjnKJeKexzlSSSgPKGBuCCkYW%2BkqEF0iWsJqS%2B-1vjXYSQlxgENa2cOxgxgdp10wABBBVIOVDNmVndoeQAGQgCAlDZTBDx8G9A2CDUABEZxXTAAEdzygJ09h5PxskkMciOcDVnAkAdYCpMBF1EQhfzAQhPTJJdnEnBV%2B0lI1MAtH0umNNl4idX1FS2KRnGTAlGiJXQZjQdAdL0dZSi4ITHmodU4jINDSAYQBeDcAcR3zj6Fd0EKM1AiCeBMAADgABjAIJcXePotOWKo9H0wwCFzUwIh8WwP0cFw3A8eKogCKBgjCdLfFiIhEhSNJFEybJcnyIpX3KLwjXIYAAApsCybBnAASkwch6FYIhSkwYAuQGpqIBagBfeM8moGA7iCD93M8rxpnoEbDN074aVheEGSWDkQvmMKVqM5FblRcyQEOKFTg2%2BlERuTZgVO87sUmYL8S%2BQkVlWEkmWO%2B7QXACFOC8Dx1B%2BwoMSpQHjnW5zLhuo67oeP7waxHbXu%2BD7qi%2BgEWWISlqUumGEWuUlyVxtkYFR7leX5QViGFd1xWNGM5QkxDVQdLUdT1DVDSZ00LU4a1bTzIjHWdV0RQ9X1sG9X1b26YMuHzZxI1IaMAWLRR42YMQUzTDMZ2UHM7REcNC04YtWMwcs9irYRa3rUNmyfYg204Dsux7IZJYHIbh1HcdMEnaVQxnOcFy6ZdV2grcIB3SVDXKXCYCPWAT2IM8LxgK8CLvB8nwkF8eLfV1P2Yb86z-HBANdkDNXAqkoJ5ODWPaRCzipXBUMwDCsK6AE8O4XPRdIyDuio-BaJEejGOY9UrfY0TMC4lPeLEAShN9ETu3ExUxSkyVZJfBSRCUz1VOIdTNLe7SMZqaZ9MMmYhD4wvC0UIkZg4bheD0jBf%2BDA4DUUAlykGENkFWdhpDAWaMQV2nAAAStRSxoTqsAAA%2Bugma7RJDeWoAAYimqQGadg5pZSCJgLq2A%2BpB1wG5Dy5DvL%2BUCsMTAI1qAjWWjMQBjh6AgAADQgCIjsZwCAADaIBJAQELjAeAuwQAAF1BEMOCKI%2BAYjQh%2BRoH5BRI0gA&features=polarisSummerEditions2023).

There's [over 6,000 usages of `<Text>` across the Admin](https://polaris-coverage.shopifycloud.com/usage/component?repo=shopify%2Fweb&type=components&element=%3CText%3E), 88 of which use `truncate`, and 172 use `alignment` - I'm not sure we're able to test all of those combinations this week :(

I'm 👍 on 80ch for long form prose, but I think it would have to be a new prop that folks can opt-in to over time.

Reply:
@jesstelford 
These are valid concerns, especially around `alignment`, however they're also supporting bad design and overall terrible typesetting in UI.
ie: aligning long lines of text (>80characters) on the right, or center, is something we just simply don't do nor should we allow it in the admin. If we're going to be offering a prop, I would do it the other way. 80ch is enforced, the prop can disable it. It shouldn't be the other way around because that would encourage bad typesetting.

Reply:
Moved discussion [to the issue](https://github.com/Shopify/polaris-summer-editions/issues/1010#issuecomment-1647000205)